src/Octoshift/GithubClient.cs

```diff
@@ -58,7 +61,7 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url)
         public virtual async Task<string> DeleteAsync(string url) => (await SendAsync(HttpMethod.Delete, url)).Content;
 
         private async Task<(string Content, KeyValuePair<string, IEnumerable<string>>[] ResponseHeaders)> SendAsync(
-            HttpMethod httpMethod, string url, object body = null)
+            HttpMethod httpMethod, string url, object body = null, bool shouldEnsureSuccessStatusCode = true)
```

Comment:
why was this flag added? We def. want to raise exceptions in case on non success responses. In case we want to treat them differently (i.e. 4xx vs 5xx) then we can raise custom exceptions based on the status code.

Reply:
We talked about it, and the api returns 302 on purpose: https://docs.github.com/en/rest/reference/migrations#download-an-organization-migration-archive